nrfcloud/apiClient.ts

```diff
@@ -136,14 +249,14 @@ export const apiClient = ({
 				}).toString()}`,
 				{ headers },
 			)
-				.then<Page<Device>>(async (res) => res.json())
-				.then((devices) => ({ devices })),
+				.then(async (res) => ({ devices: await validate(Devices, res) }))
```

Comment:
**style suggestion:** split the validate function into two, one part that checks `res.ok` and throws and error and then the validate function.

Because now you need to know that validate can trow, but the code may give the impression that it only returns. It is mixing two programming styles (throwing and returning), which I'd rather have seperated (and ideally avoid throwing all-together).

Reply:
I have separated the code where to check `response.ok` from the validation function. 

The `fetchData` function does simple job, calling `fetch` function and will throw generic error when `response.ok` is false.

Reply:
Yes, looks good. Because the *validation* happens only on a successful response.